### PR TITLE
fix: workaround ambiguous symlinks on osx

### PR DESCRIPTION
**Pull request by kormide**
This bug only surfaces on OSX. Can't reproduce on Linux.

### CODE REVIEW

File: js/private/image/index.ts [hunk 1]
```diff
@@ -39,7 +39,27 @@ function findKeyByValue(entries: Entries, value: string): string {
...
     const dirents = await readdir(dir, { withFileTypes: true })
     for (const dirent of dirents) {
-        if (dirent.isDirectory()) {
+        let isDirectory = dirent.isDirectory();
+
+        if (dirent.isSymbolicLink() && !dirent.isDirectory() && !dirent.isFile()) {
+            // On OSX we sometimes encounter this bug: https://github.com/nodejs/node/issues/30646
+            // The entry is apparently a symlink, but it's ambiguous whether it's a symlink to a
+            // file or to a directory, and lstat doesn't tell us either. Determine the type by
+            // attempting to read it as a directory.
```
Comment (gregmagolan) [inline js/private/image/index.ts:48]: woah. that is a sneaky one

**Review by thesayyn — APPROVED** [review]

**Comment by thesayyn** [comment]
it would be great to have a test for this.

**Comment by kormide** [comment]
> it would be great to have a test for this.

I agree, but since we don't run ci on macox I'm not sure the best way to do that :thinking:

**Comment by thesayyn** [comment]
right...

**Pull request merged**